powertools-parameters/powertools-parameters-appconfig/src/main/java/software/amazon/lambda/powertools/parameters/appconfig/AppConfigProviderBuilder.java

```diff
@@ -0,0 +1,120 @@
+
+public class AppConfigProviderBuilder {
+    private AppConfigDataClient client;
+    private CacheManager cacheManager;
+    private TransformationManager transformationManager;
+    private String environment;
+    private String application;
+
+    /**
+     * Create a {@link AppConfigProvider} instance.
+     *
+     * @return a {@link AppConfigProvider}
+     */
+    public AppConfigProvider build() {
+        if (cacheManager == null) {
+            cacheManager = new CacheManager();
+        }
+        if (environment == null) {
+            throw new IllegalStateException("No environment provided; please provide one");
+        }
+        if (application == null) {
+            throw new IllegalStateException("No application provided; please provide one");
+        }
+
+        // Create a AppConfigDataClient if we haven't been given one
+        if (client == null) {
+            client = AppConfigDataClient.builder()
+                    .httpClientBuilder(UrlConnectionHttpClient.builder())
+                    .region(Region.of(System.getenv(SdkSystemSetting.AWS_REGION.environmentVariable())))
+                    .overrideConfiguration(ClientOverrideConfiguration.builder()
+                            .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX,
+                                    UserAgentConfigurator.getUserAgent(BaseProvider.PARAMETERS)).build())
+                    .build();
+        }
+
+        return new AppConfigProvider(cacheManager, transformationManager, client, environment, application);
+    }
+
+    /**
+     * Set custom {@link AppConfigProvider} to pass to the {@link AppConfigDataClient}. <br/>
+     * Use it if you want to customize the region or any other part of the client.
+     *
+     * @param client Custom client
+     * @return the builder to chain calls (eg. <pre>builder.withClient().build()</pre>)
+     */
+    public AppConfigProviderBuilder withClient(AppConfigDataClient client) {
+        this.client = client;
+        return this;
+    }
+
+    /**
+     * <b>Mandatory</b>. Provide an environment to the {@link AppConfigProvider}
+     *
+     * @param environment the AppConfig environment
+     * @return the builder to chain calls (eg. <pre>builder.withCacheManager().build()</pre>)
+     */
+    public AppConfigProviderBuilder withEnvironment(String environment) {
+        this.environment = environment;
+        return this;
+    }
+
+    /**
+     * <b>Mandatory</b>. Provide an application to the {@link AppConfigProvider}
+     *
+     * @param application the application to pull configuration from
+     * @return the builder to chain calls (eg. <pre>builder.withCacheManager().build()</pre>)
+     */
+    public AppConfigProviderBuilder withApplication(String application) {
+        this.application = application;
+        return this;
+    }
+
+    /**
+     * <b>Mandatory</b>. Provide a CacheManager to the {@link AppConfigProvider}
```

Comment:
this is marked as Mandatory but if null we [create a new one](https://github.com/aws-powertools/powertools-lambda-java/pull/1403/files#diff-b307f4bdf939eae47372ead9752df4e3da4a3d0d4f360bc08619c1a59bf08663R42)

Reply:
Same for other providers. Is this really mandatory?

Reply:
Nope!